drivers/sensors/l86xxx_uorb.c

```diff
@@ -837,7 +742,7 @@ int l86xxx_register(FAR const char *uartpath, int devno)
   if (err < 0)
     {
     sensor_unreg:
-      sensor_unregister(&priv->lower, devno);
+      gnss_unregister(&priv->lower, devno);
     close_file:
       file_close(&priv->uart);
     destroy_sem:
```

Comment:
You can remove this second "if (err < 0)", just to this way:

```
  sninfo("Registered L86-XXX driver with kernel polling thread"
          " with baud rate %d and update rate %d",
          L86_XXX_BAUD_RATE,
          CONFIG_L86_XXX_FIX_INT);

    return OK;

  /* Cleanup items on error */

sensor_unreg:
  gnss_unregister(&priv->lower, devno);
close_file:
  file_close(&priv->uart);
destroy_sem:
  nxsem_destroy(&priv->run);
destroy_mutex:
  nxmutex_destroy(&priv->devlock);
free_mem:
  kmm_free(priv);
  
return err;
```


This way is more standard and easier to read